### PR TITLE
Quick Actions: Add focus border around animation panel dropdown when highlighting

### LINKED ISSUE #7626: Quick Actions: Add focus border around animation panel dropdown when highlighting
## Feature Description

The foreground image quick actions focus the animation panel when clicking the `Add animation` action. The dropdown is focused but a focus border does not appear. This action should add a focus border around the dropdown button.

## Additional Context

It works fine when focusing an input (like the `Add link` quick action). The dropdown gets focus but the `focus-visible` class doesn't get added.

no-focus

Not adding this as a bugfix since quick actions are still being built out. 

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by embarks**
## Context
- Border around the animation select button wasn't showing up when programmatically focused

## Summary
- added extra styles to the button which is focused, now it shows the ring when it is focused by the highlight functionality

## Relevant Technical Choices
- focus-visible class doesn't show up when the button is programmatically selected, so I added explicit `button:focus` styles. 
- https://github.com/WICG/focus-visible/issues/88

## User-facing changes

https://user-images.githubusercontent.com/41136059/120052883-63064e00-bfdc-11eb-9d06-e9614234c4e6.mp4

### QA

- [x] This is a non-user-facing change and requires no QA

### UAT

- [x] UAT should use the same steps as above.

### Does this PR have a security-related impact?
no

### Does this PR change what data or activity we track or use?
no

### Does this PR have a legal-related impact?
no

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #7626

### CODE REVIEW

**Review by BrittanyIRL — APPROVED** [review]
👍  bummer that the util doesn't show up for this but this is a solid work around until that issues's fixed - think you mentioned that focus-visible has a bug open to fix it?

File: assets/src/design-system/components/dropDown/select/components.js [hunk 1]
```diff
@@ -46,6 +46,13 @@ export const SelectButton = styled.button(
     text-overflow: ellipsis;
     cursor: pointer;
 
+    :focus {
```
Comment (samwhale) [inline assets/src/design-system/components/dropDown/select/components.js]: Let's add a comment here just so we know why this was added 👍

**Review by samwhale — APPROVED** [review]

**Comment by embarks** [comment]
> 👍 bummer that the util doesn't show up for this but this is a solid work around until that issues's fixed - think you mentioned that focus-visible has a bug open to fix it?

The discussion is linked in my PR summary. But from what I gathered, focus-visible wasn't really intended to fix this scenario. 

If you switch to main and use your keyboard controls to select the animation from the quick action menu, you'll notice that focus-visible gets added as intended and the focus ring shows. We're explicitly adding `:focus` styles because we _clicked_ on the quick action menu and we want the focus style to appear whether they used a keyboard or not. 

Since it was programmatically focused using a pointer, the library intentionally does not show the focus-ring on the element (i.e. it does not apply the focus-visible class). That's intended functionality. It's confusing why other buttons don't need this fix (some of them already have explicit `:focus` styles which is just from previous work) but it seemed more complicated than necessary to go down solving for consistency around other focus styles.

tl;dr it's the mixture of using a pointer and keyboard which is out of scope for the focus-visible library to really solve, so some of our buttons will want explicit `:focus` styles if they are focused programmatically with a mouse _and_ we still want the focus ring to show.

File: assets/src/design-system/components/dropDown/select/components.js (hunk 1 above)
Comment (embarks) [reply assets/src/design-system/components/dropDown/select/components.js]: Good idea, I've referenced this PR as well. I'd like to know in case there is a better solution to this. It didn't seem to affect the way the outline is applied in the Safari, Firefox, and Chrome, so it seems fine for the requirement, but maybe there is a better way to do this or indicates something we're doing wrong.

I did notice a regression, clicking on the select now activates the focus state where it didn't before. I'm going to fix that as well. Hopefully can be done with some clever css.

**Comment by embarks** [comment]
@BrittanyIRL @samwhale per the focus styles convo, I fixed the `styles.OUTLINE` and used the override prop for the effect chooser select button. I thought the prop had a very descriptive name already so I just pass it through. Let me know if I can make improvements to this that are make sense with the design system.

**Comment by embarks** [comment]
@BrittanyIRL @samwhale just tagging you as it was relevant to the inconsistent focus styles conversation.

I am now passing down highlight `styles.OUTLINE` (now using the `themeHelpers` to look more consistent) through an override prop that the `SelectButton` from the design system already had. Let me know if there's any improvements I can make there or if the override looks ok to you.

Here's the quick actions showing the ring on animations with the override:
https://user-images.githubusercontent.com/41136059/120235556-2bcbb300-c20f-11eb-9651-7253275e0ffc.mp4

And here's a couple examples of what the original implementation would have introduced:
**before**
https://user-images.githubusercontent.com/41136059/120235555-2bcbb300-c20f-11eb-92cf-41fa6e4a0ab0.mp4

**after**
https://user-images.githubusercontent.com/41136059/120235554-2b331c80-c20f-11eb-9ec0-4dbd198b798d.mp4

File: assets/src/edit-story/app/highlights/styles.js [hunk 2]
```diff
@@ -35,6 +35,5 @@ export const FLASH = css`
 `;
 
 export const OUTLINE = css`
-  box-shadow: 0 0 0 2px ${dsTheme.colors.border.focus};
-  border-radius: ${dsTheme.borders.radius.small};
+  ${({ theme }) => themeHelpers.focusCSS(theme.colors.border.focus)};
```
Comment (BrittanyIRL) [inline assets/src/edit-story/app/highlights/styles.js]: I think we should keep the border-radius here, it gets lost on things like the story title since that border isn't shown unless in an active state. And since the border.focus is the default you don't actually need to pass a color in here at all, you could just do `${themeHelpers.focusCss}`

File: assets/src/design-system/components/dropDown/select/components.js [hunk 3]
```diff
@@ -58,7 +58,7 @@ export const SelectButton = styled.button(
     css`
       ${themeHelpers.focusableOutlineCSS(
         theme.colors.interactiveBg.negativeNormal
-      )};
+      )}
```
Comment (samwhale) [inline assets/src/design-system/components/dropDown/select/components.js:61]: Do we still need this? Probably not important 😅

File: assets/src/edit-story/app/highlights/styles.js [hunk 4]
```diff
@@ -35,6 +35,6 @@ export const FLASH = css`
 `;
 
 export const OUTLINE = css`
-  box-shadow: 0 0 0 2px ${dsTheme.colors.border.focus};
+  ${themeHelpers.focusCSS};
```
Comment (samwhale) [inline assets/src/edit-story/app/highlights/styles.js:38]: 👍

**Review by samwhale — APPROVED** [review]
Nice, great update!

**Comment by samwhale** [comment]
🤔 Seems like the button gets barely squished when the focus styles appear. 
squish

File: assets/src/design-system/components/dropDown/select/components.js (hunk 3 above)
Comment (embarks) [reply assets/src/design-system/components/dropDown/select/components.js:61]: nope

**Comment by embarks** [comment]
> 🤔 Seems like the button gets barely squished when the focus styles appear.

I think i fixed it

https://user-images.githubusercontent.com/41136059/120245256-80782980-c221-11eb-9775-81d8c0052077.mp4

File: assets/src/edit-story/components/panels/design/animation/effectChooserDropdown/effectChooserDropdown.js [hunk 5]
```diff
@@ -146,6 +147,16 @@ const EffectChooserDropdown = forwardRef(function EffectChooserDropdown(
     ]
   );
 
+  const innerStyleOverrides =
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/panels/design/animation/effectChooserDropdown/effectChooserDropdown.js:150]: 👍

**Comment by samwhale** [comment]
> > 🤔 Seems like the button gets barely squished when the focus styles appear.
> 
> I think i fixed it

Yup, all fixed ✨
